src/mscorlib/corefx/System/Globalization/TextInfo.Unix.cs

```diff
@@ -36,9 +36,35 @@ private unsafe string ChangeCase(string s, bool toUpper)
             }
 
             string result = string.FastAllocateString(s.Length);
-            fixed (char* pBuf = result)
+
+            fixed (char* pResult = result)
             {
-                Interop.GlobalizationInterop.ChangeCase(s, s.Length, pBuf, result.Length, toUpper, m_needsTurkishCasing);
+                if (IsAsciiCasingSameAsInvariant && s.IsAscii())
+                {
+                    fixed (char* pSource = s)
```

Comment:
LGTM. 
do you think if we move the following code to a methods like ToUpperAsciiInvariant(string) and ToLowerAsciiInvarian(string) will be better? 
I am asking because we can optimize more (in the future) to avoid string.FastAllocateString(s.Length); if there is no casing occurring on the string. 
I am ok with the current changes though.